packages/mobile/src/invite/saga.ts

```diff
@@ -353,19 +354,39 @@ export function* withdrawFundsFromTempAccount(
   if (!fornoMode) {
     yield call(contractKit.web3.eth.personal.unlockAccount, tempAccount, TEMP_PW, 600)
   }
-  const tempAccountBalance = new BigNumber(
-    contractKit.web3.utils.fromWei(tempAccountBalanceWei.toString())
-  )
 
-  Logger.debug(TAG + '@withdrawFundsFromTempAccount', 'Creating send transaction')
-  const tx: CeloTransactionObject<boolean> = yield call(createTransaction, CURRENCY_ENUM.DOLLAR, {
+  Logger.debug(
+    TAG + '@withdrawFundsFromTempAccount',
+    `Temp account balance is ${tempAccountBalanceWei.toString()}. Calculating withdrawal fee`
+  )
+  const tempAccountBalance = divideByWei(tempAccountBalanceWei)
+  const sendTokenFeeInWei: BigNumber = yield call(getSendFee, tempAccount, CURRENCY_ENUM.DOLLAR, {
     recipientAddress: newAccount,
+    amount: tempAccountBalance,
     comment: SENTINEL_INVITE_COMMENT,
-    // TODO: appropriately withdraw the balance instead of using gas fees will be less than 1 cent
-    amount: tempAccountBalance.minus(0.01).toString(),
   })
+  const sendTokenFee = divideByWei(sendTokenFeeInWei).times(5)
+
+  if (sendTokenFee.isGreaterThanOrEqualTo(tempAccountBalance)) {
+    throw new Error('Fee is too large for amount in temp wallet')
```

Comment:
nice! this way we'll see the error more easily :+1:

Wondering, shouldn't we also check that the amount is enough to do verification as well?

Reply:
Good idea! It's a bit tough to know exactly what verification will cost but I suppose we already hard code an estimate in choosing the invite amount. I'll do this in a separate PR